File: getting-started/src/main/java/tech/donau/course/GreetingResource.java
package tech.donau.course;

import tech.donau.course.service.GreetingService;

import javax.inject.Inject;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;

@Path("/hello")
public class GreetingResource {


    @Inject
    private GreetingService greetingService;

    @GET
    @Path("/{name}")
    @Produces(MediaType.TEXT_PLAIN)
    public String hello(@PathParam("name") String name) {
        return greetingService.sayHello(name);
    }

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return greetingService.sayHello();
    }

//    public final String TOKEN = "dev";

//    @POST
//    @Path("/key")
//    @Produces(MediaType.APPLICATION_JSON)
//    @Consumes(MediaType.APPLICATION_JSON)
//    public String addHello(@HeaderParam("token") String header, @QueryParam("token") String query, String body) {
//        final String token = header != null ? header : query;
//        if (!TOKEN.equals(token)) {
//            throw new RuntimeException("Unauthorized");
//        }
//        return "{\"key\": \"" + token + "\"}";
//    }
}

File: getting-started-kotlin/src/main/kotlin/tech/donau/course/GreetingResource.kt
package tech.donau.course

import javax.inject.Inject
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.PathParam
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType

@Path("/hello")
class GreetingResource {

    @Inject
    lateinit var greetingService: GreetingService

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun hello() = "hello"

    @GET
    @Path("/greeting/{name}")
    @Produces(MediaType.APPLICATION_JSON)
    fun greet(@PathParam("name") name: String) = Greeting(greetingService.greet(name))
}